analyzers/tests/SonarAnalyzer.Test/LiveVariableAnalysis/RoslynLiveVariableAnalysisTest.FlowCaptureOperation.cs

```diff
@@ -0,0 +1,354 @@
+         *                         Exit
+         */
+        const string code = """string s = s1 ?? (s2 is null ? s3 : "NestedFalse") ?? "Hello";""";
+        var context = CreateContextCS(code, additionalParameters: "string s1, string s2, string s3");
+        context.ValidateEntry(LiveIn("s1", "s2", "s3"), LiveOut("s1", "s2", "s3"));
+        context.Validate(context.Cfg.Blocks[1], null, LiveIn("s1", "s2", "s3"), LiveOut("s1", "s2", "s3"));
+        context.Validate(context.Cfg.Blocks[2], null, LiveIn("s1", "s3"), LiveOut("s1", "s3"));
+        context.Validate(context.Cfg.Blocks[3], null, LiveIn("s1", "s2", "s3"), LiveOut("s1", "s3"));
+        context.Validate(context.Cfg.Blocks[4], null, LiveIn("s1", "s3"), LiveOut("s1", "s3"));
+        context.Validate(context.Cfg.Blocks[5], null, LiveIn("s1", "s3"), LiveOut("s1", "s3"));
+        context.Validate(context.Cfg.Blocks[6], null, LiveIn("s1", "s3"), LiveOut("s1", "s3"));
+        context.Validate(context.Cfg.Blocks[7], null, LiveIn("s1", "s3"), LiveOut("s1", "s3"));
+        context.Validate(context.Cfg.Blocks[8], null, LiveIn("s1", "s3"), LiveOut("s1", "s3"));
+        context.Validate(context.Cfg.Blocks[9], null, LiveIn("s1", "s3"));
+        context.ValidateExit();
+    }
+
+    [TestMethod]
+    public void FlowCaptrure_NullCoalescingOperator_Overwrite()
+    {
+        /*       Block 1
+         *       #0=s1
+         *         |
+         *       Block 2
+         *   #1=(s1="overwrite")
+         *     #1 is null
+         *        /   \
+         *       T     F
+         *       /     \
+         *    Block3   Block4
+         *    #2=#1   #2="value"
+         *      \         /
+         *       \       /
+         *        Block5
+         *        s1=#2
+         */
+        const string code = """s1 = (s1 = "overwrite")  ?? "value";""";
+        var context = CreateContextCS(code, additionalParameters: "string s1");
+        context.ValidateEntry(LiveIn("s1"), LiveOut("s1"));
+        context.Validate(context.Cfg.Blocks[1], null, LiveIn("s1"));
+        context.Validate(context.Cfg.Blocks[2], null);
+        context.Validate(context.Cfg.Blocks[3], null, LiveOut("s1"));
+        context.Validate(context.Cfg.Blocks[4], null, LiveOut("s1"));
+        context.Validate(context.Cfg.Blocks[5], null, LiveIn("s1"));
+        context.ValidateExit();
+    }
+
+    [TestMethod]
+    public void FlowCaptrure_SwitchStatement()
+    {
+        const string code = """
+            string result = i switch
+            {
+                0 =>  stringParam,
+                1 => "Something",
+                _ => "Everything"
+            };
+            """;
+        var context = CreateContextCS(code, additionalParameters: "int i, string stringParam");
+        context.ValidateEntry(LiveIn("i", "stringParam"), LiveOut("i", "stringParam"));
+        context.Validate(context.Cfg.Blocks[1], null, LiveIn("i", "stringParam"), LiveOut("i", "stringParam")); // if i == 0
```

Comment:
I'd do it like this, to be explicit about what you'll see during debugging. And then use #1 also below
```suggestion
        context.Validate(context.Cfg.Blocks[1], null, LiveIn("i", "stringParam"), LiveOut("i", "stringParam")); // #0 = i; if #1 == 0
```